src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,110 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private T[] arr;
+    private T[] newArr;
+    private int size = 0;
+
+    public MyLinkedList() {
+        arr = (T[]) new Object[1];
+    }
+
     @Override
     public void add(T value) {
+        if (size == arr.length) {
+            resize();
+        }
+        arr[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        newArr = (T[]) new Object[arr.length + 1];
```

Comment:
The `add(T value, int index)` method should not use an array to handle the adding process. It should properly link `Node` objects.